src/components/MMSearch/MMSearch.component.tsx

```diff
@@ -0,0 +1,199 @@
+ *
+ * @example Correct usage
+ * ```ts
+ * <MMSearch
+ *  searchValue=""
+ *  setSearchValue={(val)=> {}}
+ * 	label='label'
+ *  openOptions={true}
+ *  item={ [
+ * 		{label: 'label 1', value: 'Value 1'},
+ * 		{label: 'label 2', value: 'Value 2'},
+ * 		{label: 'label 3', value: 'Value 3'},
+ * 	]}
+ * />
+ * ```
+ *
+ * @example Correct usage for accessing the selected value
+ * ```ts
+ * <MMSearch
+ *  searchValue=""
+ *  setSearchValue={(val)=> {}}
+ * 	label='label'
+ *  optionsLoading={true}
+ *  fullWidth={true}
+ *  item={[
+ * 		{label: 'label 1', value: 'Value 1', icon: 'User'},
+ * 		{label: 'label 2', value: 'Value 2', icon: 'User'},
+ * 		{label: 'label 3', value: 'Value 3', icon: 'User'},
+ * 	]}
+ * 	onSelect={(event, options)=> {
+ * 		logic to use selected value;
+ * 	}}
+ * />
+ * ```
+ */
+export const MMSearch = (props: MMSearchProps) => {
+    const {
+        fullWidth,
+        items,
+        onSelect,
+        label,
+        className = '',
+        leadingIcon,
+        searchValue,
+        setSearchValue,
+        optionsLoading = false,
+        filterBy = '',
+        openOptions = false,
+        ...restProps
+    } = props;
+
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [open, setOpen] = useState<boolean>(openOptions);
+    const [active, setActive] = useState<number>(0);
+
+    const inputRef = useRef<HTMLInputElement>(null);
+    const listRef = useRef<HTMLUListElement>(
+        null,
+    ) as MutableRefObject<HTMLUListElement>;
+
+    /**
+     * On clicking anywhere other than `input field`, the dropdown closes
+     */
+    const onDropDownCloseHandler = (e: MouseEvent) => {
+        if (e.target !== inputRef.current) {
+            setOpen(false);
+        }
+    };
+
+    // filter dropdown options based on the input value
+    const filteredOptions = useMemo(
+        () => (filterBy ? items.filter((item) => item.label?.startsWith(filterBy) && item.label !== filterBy) : items),
+        [filterBy, items]);
+
+    useEffect(() => {
+        document.body.addEventListener('click', onDropDownCloseHandler);
+
+        return () => {
+            document.body.removeEventListener('click', onDropDownCloseHandler);
+        };
+    }, []);
+
+    /**
+     * If 'isOpen' is true and 'value' is empty, then set the active index to 0 and scroll the list to (0,0)
+     */
+    useEffect(() => {
+        if (open) {
+            setActive(0);
+            if (typeof listRef.current?.scrollTo === 'function') {
+                listRef.current.scrollTo(0, 0);
+            }
+        }
+    }, [open]);
+
+    /**
+     * The function is called when an event is detected on the keyboard,
+     * so you can browse through the list and select one.
+     */
+    const onKeyDown: React.KeyboardEventHandler<HTMLInputElement> = (event) => {
+        if (!filteredOptions.length || !open) {
+            return;
+        }
+        if (event.key === 'Enter') {
+            event.preventDefault();
```

Comment:
we can write this outside to prevent writing in each condition

Reply:
we only need this when we press enter